xbmc/addons/PVRClient.h

```diff
@@ -217,6 +217,33 @@ namespace PVR
      */
     PVR_ERROR RenameChannel(const CPVRChannelPtr &channel);
 
+    /*
+     * @brief Check if an epg tag can be recorded
+     * @param tag The epg tag
+     * @param isRecordable Set to true if the tag can be recorded
+     * @return PVR_ERROR_NO_ERROR if isRecordable was set by the addon
+     */
+    PVR_ERROR IsRecordable(const EPG::CConstEpgInfoTagPtr &tag, bool *isRecordable);
+
+    /*
+     * Check if the given EPG tag can be played.
+     * @param tag The EPG tag
+     * @return True if the EPG tag can be played
+     * @remarks Required, always return false if not supported by the addon
+     *
+     */
+    bool IsPlayable(const EPG::CConstEpgInfoTagPtr &tag);
+
+    /*
+     * Get the URL to play a given EPG tag
+     * @param tag The EPG tag
+     * @param properties The properties to be provided to the file item
+     * @return The url to play for the EPG tag, empty string if it cannot be played
+     * @remarks Required, always return empty string if not supported by the addon
+     *
+     */
+    const std::string GetEpgTagUrl(const EPG::CConstEpgInfoTagPtr &tag, const CStringPropertyMapPtr &properties);
```

Comment:
Ooops, just realized that we are in CPVRClient here. const refs are okay, then. My other comments stay valid.